app/assets/javascripts/phone_number_formatting.js.coffee

```diff
@@ -0,0 +1,37 @@
+$(document).ready ->
+  $('.phone').keydown((e) ->
+    key = e.charCode or e.keyCode or 0
+    phone = $(this)
+      
+    # Auto-format - do not expose the mask as the user begins to type
+
+    if key != 8 and key != 9
+      if phone.val().length == 4
+        phone.val phone.val() + ')'
+      if phone.val().length == 5
+        phone.val phone.val() + ' '
+      if phone.val().length == 9
+        phone.val phone.val() + '-'
+
+      # Allow numeric, tab, backspace, delete keys only
+      key == 8 or key == 9 or key == 46 or key >= 48 and
+      key <= 57 or key >= 96 and key <= 185
```

Comment:
What do these two lines do? They look like a boolean expression in the middle of nowhere (I assume it's either coffee or some weird interaction I don't know?)